python_modules/libraries/dagster-airflow/dagster_airflow/dagster_asset_factory.py

```diff
@@ -60,25 +66,28 @@ def find_upstream_dependency(node_name: str) -> None:
         for task_id in task_ids_by_asset_key[asset_key]:
             visited_nodes = {s.name: False for s in graph.nodes}
             upstream_deps = set()
-            find_upstream_dependency(f"airflow_{task_id}")
+            find_upstream_dependency(normalized_name(dag.dag_id, task_id))
             for dep in upstream_deps:
                 asset_upstream_deps.add(dep)
-            keys_by_output_name[f"result_airflow_{task_id}"] = asset_key
+            keys_by_output_name[f"result_{normalized_name(dag.dag_id, task_id)}"] = asset_key
             output_mappings.add(
                 OutputMapping(
-                    graph_output_name=f"result_airflow_{task_id}",
-                    mapped_node_name=f"airflow_{task_id}",
+                    graph_output_name=f"result_{normalized_name(dag.dag_id, task_id)}",
+                    mapped_node_name=normalized_name(dag.dag_id, task_id),
                     mapped_node_output_name="airflow_task_complete",  # Default output name
                 )
             )
 
         # the tasks for a given asset should have the same internal deps
         for task_id in task_ids_by_asset_key[asset_key]:
-            if f"result_airflow_{task_id}" in internal_asset_deps:
-                internal_asset_deps[f"result_airflow_{task_id}"].update(asset_upstream_deps)
+            if f"result_{normalized_name(dag.dag_id, task_id)}" in internal_asset_deps:
+                internal_asset_deps[f"result_{normalized_name(dag.dag_id, task_id)}"].update(
+                    asset_upstream_deps
+                )
             else:
-                internal_asset_deps[f"result_airflow_{task_id}"] = asset_upstream_deps
-            # internal_asset_deps[f"result_airflow_{task_id}"] = asset_upstream_deps
+                internal_asset_deps[
+                    f"result_{normalized_name(dag.dag_id, task_id)}"
+                ] = asset_upstream_deps
```

Comment:
I would feel marginally less terrible about this if we had a helper function to wrap all the identical f string generation

Reply:
```
def output_name_for_task_in_dag(dag_id, task_id):
   return f"result_{normalized_name(dag.dag_id, task_id)}"
```

Actually making a more proper data structure for this would be a great follow up

Reply:
Also using `@cached_method` would make me feel less bad as this is doing a regex